### DISCUSSION TITLE
Add `current_humidity` to Humidifier entity

### DISCUSSION THREAD

**Discussion by jbouwh** [home-assistant/architecture#856] [Entity Models] [answered]
The entity model for Humidifier has the `target_humidity` to set/control the desired humidity.
What misses is the current humidity and current temperature on the humidifier.
The `current_temperature` and `current_humidity` should be optional available as they are on the `climate` Entity.
The `current_temperature` is a crucial factor in calculating the humidity, and available on most humidifiers.
The `current_humidity` allows to see the difference between target and current humidity,

Update:
Suggestion is to leave `current_temperature` out of this proposal.

**Comment by frenck** [comment]
I would agree on `current_humidity`, that makes sense, as it has a `target_humidity` setting.

I think `current_temperature` should be avoided. For a climate entity, it makes sense in a way, as it sets its `target_temperature`, however, for a humidifier that is not a factor.

I don't see a good reason why an integration can't just provide a separate `sensor` entity for the temperature in those cases?

> The current_temperature is a crucial factor in calculating the humidity,

That is not a concern for the consumer of the entity, I'm not sure if I understand the reasoning behind it?

As with any proposal, it would be good to have insight into its (potential) use; which integrations would benefit from this today?

../Frenck

**Reply by MartinHjelmare** [reply]
I think these measurements should be exposed via separate sensor entities instead. That was what we said when creating the humidifier integration.

Showing the current humidity and temperature next to the settable measurements is something the frontend can solve by using the device information as context.

**Reply by jbouwh** [reply]
> I think these measurements should be exposed via separate sensor entities instead. That was what we said when creating the humidifier integration.
> 
> Showing the current humidity and temperature next to the settable measurements is something the frontend can solve by using the device information as context.

If you mean:
https://github.com/home-assistant/frontend/pull/14502

Yes, that could be nice way. of showing related sensor entities.

**Reply by MartinHjelmare** [reply]
I don't mean that PR specifically. I just mean that I think the frontend has all the information it needs to solve this already. We should not add attributes that represent sensor measurements to actuator entities when the measurements are relevant on their own.

**Comment by jbouwh** [comment]
I suggest then we add `current_humidity`, as a humidifier can set a `target_humidity` it would make sense to see the `current_humidity` (if available) directly in the humidity state card. Like for climate you don't want the `current_humidity` only as a separate sensor if is available, if you add the entity to your panel this is something different than the device state. E.g. The Xiaomi humidifier exposes both current humidity and current temperature as a separate sensor. What it misses is that you cannot directly see what you are controlling.

BTW I suggest to update the proposal and leave `current_temperature` out for now.

**Comment by piitaya** [comment]
Adding `current_humidity` can be cool for these reasons : 
- add current humidity to humidifier card (there is a [open PR](https://github.com/home-assistant/frontend/pull/14645) for this that add an adding sensor but having it as attribute can be better : )
- add current humidity to the more info dialog
- add current humidity to history graph like we have `current_temperature` for thermostat card.

We can also look up the device registry to get the current humidity value but this would not work for history graph.

And I agree, `current_temperature` should be a separate thing.

**Comment by Shulyaka** [comment]
Let me share some history on this point as it has been already discussed before.

1. In my [original proposal](https://github.com/home-assistant/architecture/issues/310#issuecomment-623507336) for the humidifiers I had suggested to add the `current_humidity` attribute along with several other attributes and services for current temperature, water level, child lock, etc.
2. Then we had some discussion and the outcome was ([confirmed](https://github.com/home-assistant/architecture/issues/310#issuecomment-627035435) by @balloob) to break everything up into separate entities linked by the device registry. I support that decision and suggest to leave it this way.

The benefits of having a separate entities are:
1. We have a unified interface to the current humidity value. As a regular sensor it is ready to use in blueprints, to see history, easier to use in automations.
2. Easier to maintain the code, all integrations using `humidifier` should not learn to use the new attribute.

The downsides that I can see are:
1. The device registry is not available for integrations that do not use config entries (i.e. those configured with yaml, such as `generic_hygrostat`). However I see it as a different problem, the `current_humidity` attribute workaround wouldn't be the proper solution for it.
2. Not so easy to add the current humidity to the hygrostat card. Again, it's a separate issue, and frontend issues should not be a justification for core changes, and I am working on a [pure frontend fix for that issue](https://github.com/home-assistant/frontend/pull/14645).

**Comment by frenck** [comment, marked as answer]
We have discussed this again today during a meeting (as we've been working on some entity dialogs).

While we have many ideas on using a more entity-level approach, which provides hints or composes devices in a way, that is all a longer-term goal at this point.

Therefore, we have decided to allow for `current_humidity` as proposed in this architectural discussion (but decline `current_temperature` for now).

../Frenck

**Reply by frenck** [reply]
Being implemented in https://github.com/home-assistant/core/pull/94874
